src/OrchardCore/OrchardCore.Users.Core/Services/ConfigureCookieAuthenticationOptions.cs

```diff
@@ -0,0 +1,33 @@
+using System.Web;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Scope;
+
+namespace OrchardCore.Users.Services;
+
+public class ConfigureCookieAuthenticationOptions : IPostConfigureOptions<CookieAuthenticationOptions>
+{
+    private readonly string _tenantName;
+
+    public ConfigureCookieAuthenticationOptions(ShellSettings shellSettings)
+    {
+        _tenantName = shellSettings.Name;
+    }
+
+    public void PostConfigure(string name, CookieAuthenticationOptions options)
+    {
+        var userOptions = ShellScope.Services.GetRequiredService<IOptions<UserOptions>>();
+
+        options.Cookie.Name = "orchauth_" + HttpUtility.UrlEncode(_tenantName);
```

Comment:
![warning-lights](https://github.com/user-attachments/assets/785a4766-2705-408f-8d64-3435522497ee)

This will badly break all OC apps using Identity (so basically most OC apps 😄), as you're now overriding the cookie name of **all the instances** of the cookies handler to share the same exact value, which means the Identity application, external, two-factor will collide, resulting in weird things like being unable to log in with an external provider.

See https://github.com/OrchardCMS/OrchardCore/issues/17631#issuecomment-2727642904 for the proper way to resolve the options.